functions/go/set-image/image_transformer.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+
+	sdk "github.com/GoogleContainerTools/kpt-functions-catalog/thirdparty/kyaml/fnsdk"
+	"sigs.k8s.io/kustomize/api/filters/imagetag"
+	"sigs.k8s.io/kustomize/api/konfig/builtinpluginconsts"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filtersutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	fnConfigGroup      = "fn.kpt.dev"
+	fnConfigVersion    = "v1alpha1"
+	fnConfigAPIVersion = fnConfigGroup + "/" + fnConfigVersion
+	fnConfigKind       = "SetImage"
+)
+
+// getDefaultImageFields returns default image FieldSpecs
+func getDefaultImageFields() (types.FsSlice, error) {
+	type defaultConfig struct {
+		FieldSpecs types.FsSlice `json:"images,omitempty" yaml:"images,omitempty"`
+	}
+	defaultConfigString := builtinpluginconsts.GetDefaultFieldSpecsAsMap()["images"]
+	var tc defaultConfig
+	err := yaml.Unmarshal([]byte(defaultConfigString), &tc)
+	return tc.FieldSpecs, err
+}
+
+// validGVK returns whether the given sdk.KubeObject is of the given apiVersion/kind
+func validGVK(ko *sdk.KubeObject, apiVersion, kind string) bool {
+	return ko.APIVersion() == apiVersion && ko.Kind() == kind
+}
+
+type SetImage struct {
+	// Image Desired image tag.
+	Image types.Image `json:"image,omitempty" yaml:"image,omitempty"`
+	// AdditionalImageFields is used to specify additional fields to set image.
+	AdditionalImageFields types.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+}
+
+// validateInput validates the inputs passed into via the functionConfig
+func (si *SetImage) validateInput() error {
+	if si.Image.Name == "" {
+		return fmt.Errorf("missing image name")
+	}
+	if si.Image.NewName == "" && si.Image.NewTag == "" && si.Image.Digest == "" {
+		return fmt.Errorf("missing image newName, newTag, or digest")
+	}
+	if si.Image.NewTag != "" && si.Image.Digest != "" {
+		return fmt.Errorf("image newTag and digest both set")
+	}
+	return nil
+}
+
+// Config initializes SetImage from a functionConfig sdk.KubeObject
+func (si *SetImage) Config(functionConfig *sdk.KubeObject) error {
+	si.Image = types.Image{}
+	si.AdditionalImageFields = nil
+	switch {
+	case validGVK(functionConfig, "v1", "ConfigMap"):
+		functionConfig.GetOrDie(&si.Image, "data")
+	case validGVK(functionConfig, fnConfigAPIVersion, fnConfigKind):
+		functionConfig.AsOrDie(si)
```

Comment:
I'd suggest not suggest using *OrDie in our catalog. Reasons are we can't guarantee that it will always succeed and we should be more conservative. And we should provide a little more context in the error e.g. "unable to get field data: %w" and "unable to convert the functionConfig to ...: %w".

Reply:
Good suggestion - done